File: PN13.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

   

    public int SumFactors()
    {
            int iSum = 0;
            int iCnt = 0;

            for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
            {
                if((iNo % iCnt) == 0)
				{
					iSum = iSum + iCnt;
				}
            }
            return iSum;
    }
}

class PN13
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;

            nobj.Accept();
          

            iRet =nobj.SumFactors();
            System.out.println("Sum of factor is : "+iRet);
    }
}

File: PN13.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void Display()      
        {
            cout<<"Value is : "<<this->iNo<<endl;
        }

        void DisplayFactors()
		{
			int iCnt = 0;
			
			for(iCnt = 1;iCnt < iNo; iCnt++)
			{
				if((iNo % iCnt) == 0)
				{
					cout<<iCnt<<endl;
				}
			}
		
		}

};

int main()
{
    Number nobj;

    nobj.Accept();
    nobj.Display();

    nobj.DisplayFactors();

    return 0;
}